torch/utils/_pytree.py

```diff
@@ -283,11 +304,50 @@ def register_dataclass(cls: type[Any]) -> None:
         >>> assert torch.allclose(point.y, torch.tensor(2))
 
     """
-    import torch.export
+    if not dataclasses.is_dataclass(cls):
+        assert field_names is not None, (
+            "field_names must be specified with a list of all"
+            "attribute names if cls is not a dataclass"
+        )
+    else:
+        dataclass_fields = [f.name for f in dataclasses.fields(cls)]
```

Comment:
```suggestion
        dataclass_fields = [f.name for f in dataclasses.fields(cls) if f.init]
```

We should only save the `init` parameters for `cls(**dict(zip(field_names, values)))`. We'd better also add a test for this.

Reply:
Thanks! sorry for the delay! I updated the code to include "drop_field_names"